src/javascript/app_2/components/elements/data_table.jsx

```diff
@@ -166,40 +36,61 @@ class DataTable extends React.Component {
     }
 
     renderBodyRows() {
-        return this.state.display_data.map((transaction, id) => this.renderRow(transaction, id));
+        return this.props.data_source
+            .map((transaction, id) => this.renderRow(transaction, id));
     }
 
     renderHeaders() {
-        return this.props.columns.map(col => <th key={col.data_index}>{col.title}</th>);
+        return this.props.columns.map(col => <th className={col.data_index} key={col.data_index}>{col.title}</th>);
     }
 
-    render() {
-        const { pagination } = this.props;
+    renderTableClone() {
+        {/*
+            cloned table with one row for fixed header
+            inspired by
+            https://stackoverflow.com/questions/4709390
+        */}
+        return (
+            <table className='table table-clone'>
+                <thead className='table-head'>
+                    <tr className='table-row'>
+                        {this.renderHeaders()}
+                    </tr>
+                </thead>
+
+                <tbody className='table-body'>
+                    {this.renderRow(this.props.data_source[0], 0)}
+                </tbody>
+            </table>
+        );
+    }
 
+    render() {
         return (
-            <div className='table-container'>
+            <div className='table-container' ref={this.props.has_fixed_header && this.fixHeaderInPlace}>
+                <div className='table-filter'>
+                    two datepickers
+                </div>
                 <table className='table'>
-                    <thead className='table-thead'>
+                    <thead className='table-head'>
                         <tr className='table-row'>
                             {this.renderHeaders()}
                         </tr>
                     </thead>
 
-                    <tbody className='table-tbody'>
+                    <tbody className='table-body'>
                         {this.renderBodyRows()}
                     </tbody>
                 </table>
 
-                {pagination && this.renderPagination()}
+                {this.props.has_fixed_header && this.renderTableClone()}
             </div>
         );
     }
 }
 
 DataTable.defaultProps = {
-    pagination        : true,
-    page_size         : 10,
-    pages_close_to_end: 5,
+    has_fixed_header: false,
```

Comment:
Is this needed since it initially is false?

Reply:
good point, `undefined` will do as well :D